### PR TITLE
Feature routes controllers

### PR DESCRIPTION
**Pull request by snjnlsn**
@markfranciose @mjcataldi

### CODE REVIEW

File: db/migrate/20170818172013_create_habits.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class CreateHabits < ActiveRecord::Migration[5.1]
+habclass CreateHabits < ActiveRecord::Migration[5.1]
```
Comment (markfranciose) [inline db/migrate/20170818172013_create_habits.rb:1]: Just have to delete hab here. @sanjaynelson  can you do that on your branch and re-push?

File: app/models/habit.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Habit < ApplicationRecord
```
Comment (markfranciose) [inline app/models/habit.rb:1]: This model looks pro

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+RSpec.describe UsersController, type: :controller do
+
+  context "when the user is not valid" do
+
+    it "returns a status code of 422 for an invalid user" do
+         post :create, params: { user: { user_identifier: nil } }
+         expect(response.status).to eq 422
+    end
+
+    it "returns a json object with errors in response body" do
```
Comment (markfranciose) [inline spec/controllers/users_controller_spec.rb:12]: nice - testing JSON objects w/ error messages

**Review by markfranciose — COMMENTED** [review]
Looks awesome, just need to change the error in the habit migration. Controllers and models look super pro.

File: app/controllers/habits_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
+class HabitsController < ApplicationController
+  before_action :set_habit, only: [:show, :update]
+  before_action :set_user, only: [:index, :create]
+
+  def index
+    if @user
+      render json: @user.habits, status: 201
```
Comment (mjcataldi) [inline app/controllers/habits_controller.rb]: ## Status Code

Everything looks great, good work guys!  In general, the only thing I would change is the status codes of the index and show actions in the HabitsController.  From what I've been reading, status 201 is the code attributed to the successful creation of a record vs. 200 which is a successful GET request.  Also, 422 is the client-side error equivalent.  My first instinct is to label the show error as a 400, but that may be incorrect.

https://httpstatuses.com/
@markfranciose @sanjaynelson @tony-rodriguez

**Pull request merged**